src/Data/List/ZipList.purs

```diff
@@ -25,6 +28,24 @@ newtype ZipList a = ZipList (List a)
 runZipList :: forall a. ZipList a -> List a
 runZipList (ZipList xs) = xs
 
+-- | Transpose any structure which is both `Foldable` and `Unfoldable`,
+-- | via `ZipList`; note that `sequence` for `ZipList` is the same as
+-- | `transpose`.
+transpose :: forall f a. (Foldable f, Unfoldable f) => f (f a) -> f (f a)
+transpose =
```

Comment:
Given that I don't have to worry about the core library dependency graph, I won't make any suggestions about where I think this should go, I'll leave that up to you ;)

I will say, though, that I don't see how this could go into `foldable-traversable` or `unfoldable` (at least, not without duplicating parts of `lists`) because the implementation does depend on the `Traversable ZipList` instance, and we can't depend on `ZipList` in `foldable-traversable` or `unfoldable` because `lists` already depends on both of those.

Maybe we should just start with a version defined in terms of `List` and not `Foldable`, then, and see what happens.

Come to think of it, why does `ZipList` wrap lazy lists and not strict ones? Would it make sense to have a zippy strict list newtype too?